File: Codewars/Python/7 kyu/Going to the cinema.py
import math
def movie(card, ticket, perc):
    planB = card+ticket*perc
    cont=1
    while math.ceil(planB)>ticket*cont:
        planB+= ticket*perc**cont*perc
        cont+=1
    return cont

print(movie(500, 15, 0.9))

File: Codewars/Javascript/7 kyu/I/Going to the cinema.js
function movie(card, ticket, perc) {
    let planB=card+ticket*perc, cont=1;
    while(Math.ceil(planB)>ticket*cont){
        cont++
        planB+=ticket*Math.pow(perc,cont)*perc
    }
    return cont+1
};

const movieUp = (card, ticket, perc) =>{
    var costB = card, costA=0, n=0;
    while(Math.ceil(costB)>= costA){
        costA +=ticket;
        costB += ticket * Math.pow(perc, ++n)
    }
    return n
}

console.log(movie(500, 15, 0.9));
console.log(movieUp(500, 15, 0.9));